.github/workflows/src/sdk-breaking-change-labels.js

```diff
@@ -0,0 +1,186 @@
+              labelName = sdkLabels[`${language}`].breakingChange;
+            }
+          } else {
+            core.error(`Failed to fetch artifacts: ${response.status}, ${response.statusText}`);
+            const errorText = await response.text();
+            core.error(`Error details: ${errorText}`);
+          }
+        } catch (apiError) {
+          core.error(`Error calling Azure DevOps API: ${apiError}`);
+        }
+      } else {
+        core.error(`Could not extract organization and project from URL: ${checkRun.details_url}`);
+      }
+    } else {
+      core.error(`Could not extract buildId from details_url: ${checkRun.details_url}`);
+    }
+  } else {
+    core.info('No details_url found in check run');
+  }
+  // Get the issue number from the check run
+  if (Number.isNaN(issue_number)) {
+    const { issueNumber } = await getIssueNumber({checkRun, core, github});
+    issue_number = issueNumber;
+  }
+
+  if (!labelAction) {
+    core.info('No label action found, defaulting to None');
+    labelAction = LabelAction.None;
+  }  
+  
+  return { labelName, labelAction, issueNumber: issue_number, state: labelAction !== LabelAction.None ? 'success' : 'neutral' };  
+}
+
+/**
+ * Retrieves the PR number associated with a specific commit SHA
+ * @param {Object} params
+ * @param {import('github').CheckRun} params.checkRun - The check run object containing head_sha
+ * @param {typeof import("@actions/core")} params.core - GitHub Actions core for logging
+ * @param {import('github-script').GitHub} params.github - GitHub API client
+ * @returns {Promise<{issueNumber: number}>} - The PR number or NaN if not found
+ */
+async function getIssueNumber({checkRun, core, github})
+{
+  let issueNumber = NaN;
+
+  if (!checkRun.head_sha) {
+    core.info('No head_sha found in check run');
+    return { issueNumber };
+  }
+
+  core.info(`Searching for PRs with commit SHA: ${checkRun.head_sha}`);
+
+  try {
+    const searchResponse = await github.rest.search.issuesAndPullRequests({
+      q: `sha:${checkRun.head_sha} type:pr state:open`
+    });
+
+    const totalCount = searchResponse.data.total_count;
+    const itemsCount = searchResponse.data.items.length;
+
+    core.info(`Search results: ${totalCount} total matches, ${itemsCount} items returned`);
+
+    if (itemsCount > 0) {
+      const firstItem = searchResponse.data.items[0];
+      issueNumber = firstItem.number;
+      core.info(`Found the first matched PR #${issueNumber}: ${firstItem.html_url}`);
+
+      if (itemsCount > 1) {
+        core.warning(`Multiple PRs found for commit ${checkRun.head_sha}: ${searchResponse.data.items.map(item => `#${item.html_url}`).join(', ')}`);
+      }
+    } else {
+      core.info(`No open PRs found for commit ${checkRun.head_sha}`);
+    }
+  } catch (error) {
+    core.error(`${wfName}: Error searching for PRs with commit ${checkRun.head_sha}: ${error.message}`);
+    throw error;
+  }
+
+  return { issueNumber };
+}
```

Comment:
Add trailing newline.  Your IDE might be removing this.

Reply:
Added.